entity-management-definitions/src/main/java/eu/europeana/entitymanagement/normalization/EntityFieldsValidator.java

```diff
@@ -0,0 +1,299 @@
+                            "A-Z]{2,7}$";                              
+        Pattern pat = Pattern.compile(emailRegex);
+        if (fieldValue == null) return false;
+        if (pat.matcher(fieldValue.toString()).matches()) {
+        	return true;
+        }
+        else
+        {
+    		addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type Email and contains inappropriate characters: "+fieldValue.toString()+".");
+    		return false;
+        }
+    }
+    
+    boolean checkDateFormatISO8601(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+    	try {
+        	//LocalDate.parse(fieldValue.toString(), DateTimeFormatter.ofPattern(java.time.format.DateTimeFormatter.ISO_DATE.toString()).withResolverStyle(ResolverStyle.STRICT));
+    		LocalDate.parse(fieldValue.toString(), java.time.format.DateTimeFormatter.ISO_DATE);
+        } catch (DateTimeParseException e) {
+    		addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type Date and does not comply with the ISO-8601 format: "+fieldValue.toString()+".");
+            returnValue = false;
+        }
+    	return returnValue;
+    }
+    
+    boolean checkURIFormat(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue, boolean shouldBeURI) {
+    	if (shouldBeURI && !EntityUtils.isUri(fieldValue.toString())) {
+			addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type: "+EntityFieldsTypes.valueOf(field.getName()).getFieldType()+" and it's element: "+fieldValue.toString()+" does not have the URI form.");
+			returnValue = false;
+		}
+    	else if (!shouldBeURI && EntityUtils.isUri(fieldValue.toString())) {
+			addConstraint(returnValue, context, "The entity field: "+field.getName()+" is of type: "+EntityFieldsTypes.valueOf(field.getName()).getFieldType()+" and it's element: "+fieldValue.toString()+" has the URI form.");
+			returnValue = false;
+		}
+    	return returnValue;
+    }
+   
+    boolean validateTextOrKeywordField (ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+    	//non-multilingual fields are not of type Map, but they can be of type String[], List<String>, or String
+    	if (!EntityFieldsTypes.valueOf(field.getName()).getFieldIsmultilingual()) {
+    		returnValue = validateNonMultilingualTextOrKeywordField(context, field, fieldValue, returnValue);    		
+    	}
+    	else {
+    		returnValue = validateMultilingualTextOrKeywordField(context, field, fieldValue, returnValue); 
+    		returnValue = validateLanguageCodes(context, field, fieldValue, returnValue);    		
+    	}
+    	return returnValue;
+    }
+    
+    boolean validateNonMultilingualTextOrKeywordField(ConstraintValidatorContext context, Field field, Object fieldValue, boolean returnValue) {
+    	//if the field is an array
+		if (field.getType().isArray()) {
```

Comment:
the fields were recently changed to List<String> instead of String[], in any case, the same method should be used for validation of string arrays and lists, by simply using Arrays.asList(String[]). Please change to this behaviour in all methods that handle string arrays

Reply:
Improved.